akka-docs/src/main/paradox/typed/persistence-durable-state.md

```diff
@@ -0,0 +1,391 @@
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #actor-context }
+
+## Changing Behavior
+
+After processing a message, actors are able to return the `Behavior` that is used
+for the next message.
+
+As you can see in the above examples this is not supported by persistent actors. Instead, the state is
+persisted as an `Effect` by the `commandHandler`. 
+
+The reason a new behavior can't be returned is that behavior is part of the actor's
+state and must also carefully be reconstructed during recovery from the persisted state. This would imply
+that the state needs to be encoded such that the behavior can also be restored from it. 
+That would be very prone to mistakes and thus not allowed in Akka Persistence.
+
+For basic actors you can use the same set of command handlers independent of what state the entity is in.
+For more complex actors it's useful to be able to change the behavior in the sense
+that different functions for processing commands may be defined depending on what state the actor is in.
+This is useful when implementing finite state machine (FSM) like entities.
+
+The next example demonstrates how to define different behavior based on the current `State`. It shows an actor that
+represents the state of a blog post. Before a post is started the only command it can process is to `AddPost`.
+Once it is started then one can look it up with `GetPost`, modify it with `ChangeBody` or publish it with `Publish`.
+
+The state is captured by:
+
+Scala
+:  @@snip [BlogPostEntityDurableState.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BlogPostEntityDurableState.scala) { #state }
+
+Java
+:  @@snip [BlogPostEntityDurableState.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BlogPostEntityDurableState.java) { #state }
+
+The commands, of which only a subset are valid depending on the state:
+
+Scala
+:  @@snip [BlogPostEntityDurableState.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BlogPostEntityDurableState.scala) { #commands }
+
+Java
+:  @@snip [BlogPostEntityDurableState.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BlogPostEntityDurableState.java) { #commands }
+
+@java[The command handler to process each command is decided by the state class (or state predicate) that is
+given to the `forStateType` of the `CommandHandlerBuilder` and the match cases in the builders.]
+@scala[The command handler to process each command is decided by first looking at the state and then the command.
+It typically becomes two levels of pattern matching, first on the state and then on the command.]
+Delegating to methods is a good practice because the one-line cases give a nice overview of the message dispatch.
```

Comment:
Add here some reference to `addPost` and other methods to be clear

Reply:
done ..